rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeperServer.java

```diff
@@ -82,9 +83,32 @@ public void run()
     return 0;
   }
 
-  public void startServer(Configuration conf, MetricRegistry metricsRegistry)
+  public BookKeeper startServer(final Configuration conf, MetricRegistry metricsRegistry)
   {
-    BookKeeper localBookKeeper;
+    setupServer(conf, metricsRegistry);
+    if (CacheConfig.isEmbeddedModeEnabled(conf)) {
+      new Thread(new Runnable()
+      {
+        @Override
+        public void run()
+        {
+          startThriftServer(conf, localBookKeeper);
+        }
+      }).start();
+    }
+    else {
+      startThriftServer(conf, localBookKeeper);
+    }
+
+    return localBookKeeper;
+  }
+
+  public void setupServer(Configuration conf, MetricRegistry metricsRegistry)
+  {
+    conf = new Configuration(conf);
```

Comment:
Why are we creating new conf object?

Reply:
So that any changes made in the conf object inside BKS are not reflected in client side. In case of embedded mode with FileSystem cache enabled, this can disable Rubix for client if we use the same conf object passed down.